sound/pci/hda/patch_realtek.c

```diff
@@ -4901,6 +4903,8 @@ static void alc_update_headset_mode(struct hda_codec *codec)
 	switch (new_headset_mode) {
 	case ALC_HEADSET_MODE_UNPLUGGED:
 		alc_headset_mode_unplugged(codec);
+		spec->current_headset_mode = ALC_HEADSET_MODE_UNKNOWN;
```

Comment:
spec->current_headset_mode will be set to new_headset_mode later in this function. So it seems not necessary to set spec->current_headset_mode = ALC_HEADSET_MODE_UNKNOWN. But the code should be harmless here. So this is fine to me.